smtp/response.js

```diff
@@ -48,46 +66,74 @@ class SMTPResponse {
 			);
 		};
 
+		/**
+		 * @param {string | Buffer} data the data
+		 * @returns {void}
+		 */
 		const watch = data => {
 			if (data !== null) {
 				buffer += data.toString();
 				notify();
 			}
 		};
 
+		/**
+		 * @param {Error} err the error object
+		 * @returns {void}
+		 */
 		const close = err => {
 			stream.emit(
 				'response',
 				SMTPError('connection has closed', SMTPError.CONNECTIONCLOSED, err)
 			);
 		};
 
+		/**
+		 * @param {Error} err the error object
+		 * @returns {void}
+		 */
 		const end = err => {
 			stream.emit(
 				'response',
 				SMTPError('connection has ended', SMTPError.CONNECTIONENDED, err)
 			);
 		};
 
+		/**
+		 * @param {Error} [err] the error object
+		 * @returns {void}
+		 */
 		this.stop = err => {
 			stream.removeAllListeners('response');
-			stream.removeListener('data', watch);
-			stream.removeListener('end', end);
-			stream.removeListener('close', close);
-			stream.removeListener('error', error);
+			if (stream instanceof TLSSocket) {
+				stream.removeListener('data', watch);
+				stream.removeListener('end', end);
+				stream.removeListener('close', close);
+				stream.removeListener('error', error);
+			}
 
-			if (err && typeof onerror === 'function') {
+			if (err != null && typeof onerror === 'function') {
```

Comment:
i'm being fastidious here, since this pattern produces more efficient code. if it needs to be a more comprehensive existential check i can change it back.

Reply:
what about undefined?

Reply:
`!= null` catches both `null` and `undefined` (and nothing else) -- cf. [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Loose_equality_using)